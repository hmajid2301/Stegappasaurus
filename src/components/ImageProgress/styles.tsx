import { Dimensions, StyleSheet } from "react-native";

const pageWidth = Dimensions.get("window").width;

const styles = StyleSheet.create({
  progressContainer: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center"
  },

  circularImage: {
    borderRadius: pageWidth * 0.75,
    borderWidth: 5,
    overflow: "hidden"
  },

  image: {
    height: pageWidth * 0.75,
    opacity: 0.65,
    width: pageWidth * 0.75
  }
});

export default styles;
