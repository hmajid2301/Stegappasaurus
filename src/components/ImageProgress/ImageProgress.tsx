import React from 'react';
import {
  Dimensions,
  ImageBackground,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import {Icon, IconType} from 'react-native-elements';
import PercentageCircle from 'react-native-percentage-circle';

import {TabColors, ThemeColors} from '~/constants/types';
import styles from './styles';

interface IProps {
  background: ThemeColors;
  onPress?: (...args: any) => void;
  icon?: {
    color: string;
    name: string;
    size: number;
    type: IconType;
  };
  progress: number;
  photo: string;
  primaryColor: TabColors;
}

const pageWidth = Dimensions.get('window').width;

const ImageProgress: React.FunctionComponent<IProps> = (props: IProps) => (
  <View style={[styles.progressContainer, {backgroundColor: props.background}]}>
    <TouchableOpacity activeOpacity={0.8} onPress={props.onPress}>
      <PercentageCircle
        borderWidth={5}
        color={props.primaryColor}
        percent={props.progress}
        radius={pageWidth * 0.334}>
        <ImageBackground
          imageStyle={[styles.circularImage, {borderColor: props.primaryColor}]}
          source={{uri: props.photo}}
          style={styles.image}>
          <View style={styles.iconContainer}>
            {props.icon && <Icon {...props.icon} />}
            <Text style={styles.textPercentage}>{props.progress}%</Text>
          </View>
        </ImageBackground>
      </PercentageCircle>
    </TouchableOpacity>
  </View>
);

export default ImageProgress;
