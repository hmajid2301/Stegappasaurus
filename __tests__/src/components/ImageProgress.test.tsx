import { shallow } from "enzyme";
import React from "react";

import ImageProgress from "~/components/ImageProgress";

describe("ImageProgress: Match snapshots", () => {
  test("1", () => {
    const component = shallow(
      <ImageProgress
        animating={true}
        background="#17212D"
        icon={{
          color: "#FFF",
          name: "share",
          size: 130,
          type: "font-awesome"
        }}
        onPress={() => null}
        photo="images.png"
        progress={100}
        primaryColor="#009CFF"
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <ImageProgress
        animating={true}
        background="#FFF"
        icon={{
          color: "#111",
          name: "share",
          size: 100,
          type: "font-awesome"
        }}
        onPress={() => null}
        photo="images.png"
        progress={0}
        primaryColor="#E88C0C"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
