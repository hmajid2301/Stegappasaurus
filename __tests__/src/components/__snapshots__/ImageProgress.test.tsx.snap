// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageProgress: Match snapshots 1 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#17212D",
      },
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <PercentageCircle
      bgcolor="#e3e3e3"
      borderWidth={5}
      color="#009CFF"
      innerColor="#fff"
      percent={100}
      radius={250.50000000000003}
    >
      <ImageBackground
        imageStyle={
          Array [
            Object {
              "borderRadius": 562.5,
              "borderWidth": 5,
              "overflow": "hidden",
            },
            Object {
              "borderColor": "#009CFF",
            },
          ]
        }
        source={
          Object {
            "uri": "images.png",
          }
        }
        style={
          Object {
            "height": 562.5,
            "opacity": 0.65,
            "width": 562.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Themed.Icon
            color="#FFF"
            name="share"
            size={130}
            type="font-awesome"
          />
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Roboto",
                "fontSize": 50,
              }
            }
          >
            100
            %
          </Text>
        </View>
      </ImageBackground>
    </PercentageCircle>
  </TouchableOpacity>
</View>
`;

exports[`ImageProgress: Match snapshots 2 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <PercentageCircle
      bgcolor="#e3e3e3"
      borderWidth={5}
      color="#E88C0C"
      innerColor="#fff"
      percent={0}
      radius={250.50000000000003}
    >
      <ImageBackground
        imageStyle={
          Array [
            Object {
              "borderRadius": 562.5,
              "borderWidth": 5,
              "overflow": "hidden",
            },
            Object {
              "borderColor": "#E88C0C",
            },
          ]
        }
        source={
          Object {
            "uri": "images.png",
          }
        }
        style={
          Object {
            "height": 562.5,
            "opacity": 0.65,
            "width": 562.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Themed.Icon
            color="#111"
            name="share"
            size={100}
            type="font-awesome"
          />
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Roboto",
                "fontSize": 50,
              }
            }
          >
            0
            %
          </Text>
        </View>
      </ImageBackground>
    </PercentageCircle>
  </TouchableOpacity>
</View>
`;
